src/vector.rs

```diff
@@ -0,0 +1,84 @@
+use mlua::prelude::LuaTable;
+use mlua::{Lua, Result};
+
+use crate::world::types::Location;
+
+pub fn sign(x: f64) -> f64 {
+    if x < 0.0 {
+        -1.0
+    } else if x > 0.0 {
+        1.0
+    } else {
+        0.0
+    }
+}
+
+pub fn magnitude(x: f64, y: f64) -> f64 {
+    f64::sqrt(x * x + y * y)
+}
+
+pub fn angle(x: f64, y: f64) -> f64 {
+    f64::atan2(y, x)
+}
+
+pub fn components(r: f64, angle: f64) -> (f64, f64) {
+    let x: f64 = r * f64::cos(angle);
+    let y: f64 = r * f64::sin(angle);
+
+    (x, y)
+}
+
+pub fn rotate(x: f64, y: f64, angle: f64) -> (f64, f64) {
+    let r: f64 = magnitude(x, y);
+    let t: f64 = self::angle(x, y);
+    components(r, t + angle)
+}
+
+pub fn add(a: Location, b: Location) -> Location {
+    let mut l = Location(0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
+    (l.0, l.1) = rotate(b.0, b.1, b.2);
+    l.2 = b.2;
+
+    (l.3, l.4) = rotate(b.3, b.4, b.5);
+    l.5 = b.5;
```

Comment:
The Lua code used to conditionally check for the second half of the location (the x, y, and angle velocities). But in Rust we can guarantee that those values are always filled in with either 0.0 or an explicit value.